// react-google-login-ng/dist/react-google-login-ng.esm.js
import React, { useEffect } from 'react';

function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;
  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }
  return target;
}

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = "#NG_GLOGIN_BTN{position:relative;-webkit-user-select:none;-moz-user-select:none;user-select:none;width:-moz-fit-content;width:fit-content}#NG_GLOGIN_BTN.disabled:after{background-color:hsla(0,0%,100%,.277);content:\" \";cursor:not-allowed;height:100%;left:0;position:absolute;top:0;width:100%}";
styleInject(css_248z);

var _excluded = ["disabled", "script", "successCallback", "config", "method"];
var GButton = function GButton(_ref) {
  var disabled = _ref.disabled,
    _ref$script = _ref.script,
    src = _ref$script === void 0 ? 'https://accounts.google.com/gsi/client' : _ref$script,
    successCallback = _ref.successCallback,
    config = _ref.config,
    method = _ref.method,
    props = _objectWithoutPropertiesLoose(_ref, _excluded);
  var BTN_CONTAINER = "NG_GLOGIN_BTN";
  useEffect(function () {
    var _document$querySelect;
    var initializeGsi = function initializeGsi() {
      try {
        var _window$google;
        (_window$google = window.google) == null ? void 0 : _window$google.accounts.id.initialize(_extends({
          callback: function callback(resp) {
            successCallback(_extends({}, resp, {
              token_info_url: "https://oauth2.googleapis.com/tokeninfo?id_token=" + resp.credential
            }));
            // Register state in cookies when user leave of site
            window.google.accounts.id.disableAutoSelect();
            //Cancel One Tap FLow
            window.google.accounts.id.cancel();
          }
        }, props));
        if (method !== "one_tap") {
          // Show button
          window.google.accounts.id.renderButton(document.getElementById(BTN_CONTAINER), config || {});
        }
        if (method !== "button" && !disabled) {
          // Show One Tap
          window.google.accounts.id.prompt(function () {});
        }
      } catch (error) {
        console.log({
          error: error
        });
      }
    };
    var script = document.createElement('script');
    script.src = src;
    script.async = true;
    script.onload = initializeGsi;
    (_document$querySelect = document.querySelector('body')) == null ? void 0 : _document$querySelect.appendChild(script);
    return function () {
      var _document$querySelect2;
      (_document$querySelect2 = document.querySelector('body')) == null ? void 0 : _document$querySelect2.removeChild(script);
    };
  }, []);
  useEffect(function () {
    if (disabled) {
      var _window$google2;
      (_window$google2 = window.google) == null ? void 0 : _window$google2.accounts.id.cancel();
    }
  }, [disabled]);
  return React.createElement("div", {
    id: BTN_CONTAINER,
    className: disabled ? "disabled" : ""
  }, method !== "one_tap" && React.createElement("small", null, "Loading"));
};

var GoogleLogin = GButton;
// import Form from './components/FormElements/Form';

export default GoogleLogin;
